egs2/TEMPLATE/asr1/pyscripts/utils/plot_sinc_filters.py

```diff
@@ -0,0 +1,318 @@
+            plt.xlabel("kernel index")
+            plt.plot(x, kernel)
+            plt.plot(x, pre_kernel, "--", alpha=0.5)
+            img_name = "filter_kernel_both_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = str(args.out_folder / img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            y = np.zeros_like(x_f)
+            y[F_mins[i] : F_maxs[i]] = 1.0
+            plt.clf()
+            plt.plot(x_f, y)
+            img_name = "filter_freq_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = str(args.out_folder / img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            pre_y = np.zeros_like(x_f)
+            pre_y[pre_F_mins[i] : pre_F_maxs[i]] = 1.0
+            plt.clf()
+            plt.plot(x_f, y)
+            plt.plot(x_f, pre_y)
+            img_name = "filter_freq_both_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = args.out_folder / img_name
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+    plt.clf()
+    filters = [32, 71, 113, 126]
+    fig, axs = plt.subplots(2, 2, sharex=True, sharey="row")
+
+    axs[0, 0].plot(x, kernels[filters[0]][0])
+    axs[0, 0].plot(x, pre_kernels[filters[0]][0], "--", alpha=0.5)
+    axs[0, 1].plot(x, kernels[filters[1]][0])
+    axs[0, 1].plot(x, pre_kernels[filters[1]][0], "--", alpha=0.5)
+    axs[1, 0].plot(x, kernels[filters[2]][0])
+    axs[1, 0].plot(x, pre_kernels[filters[2]][0], "--", alpha=0.5)
+    axs[1, 1].plot(x, kernels[filters[3]][0])
+    axs[1, 1].plot(x, pre_kernels[filters[3]][0], "--", alpha=0.5)
+
+    img_name = "filter_kernel_ensemble2.%s" % (args.filetype)
+    img_path = str(args.out_folder / img_name)
+    plt.savefig(img_path, bbox_inches="tight")
+    plt.close(fig)
+    print("Plotted %s" % img_path)
+
+
+def plot_filters(indices, filename, F_mins, F_maxs, output_folder):
+    """Plot filters bandwidths.
+
+    Args:
+        indices: Sorted indices of filters.
+        filename: Output filename (png or svg).
+        F_mins: Minimum frequencies.
+        F_maxs: Maximum frequencies.
+        output_folder: Output folder.
+    """
+    x = np.linspace(0, np.max(F_maxs), np.max(F_maxs) + 1)
+    plt.clf()
+    height = 1
+    for i in indices:
+        y = np.zeros_like(x)
+        y[F_mins[i] : F_maxs[i]] = height
+        height += 1
+        plt.plot(x, y)
+    img_path = str(output_folder / filename)
+    plt.savefig(img_path, bbox_inches="tight")
+    print("Plotted %s" % img_path)
+
+
+def main(argv):
+    """Load the model, generate kernel and bandpass plots."""
+    parser = get_parser()
+    args = parser.parse_args(argv)
+    model_path = args.model_path
+    sample_rate = args.sample_rate
+
+    model = torch.load(model_path, map_location="cpu")
+    if "model" in model:  # snapshots vs. model.acc.best
```

Comment:
```suggestion
    if isinstance(model, dict) and "model" in model:  # snapshots vs. model.acc.best
```

Now this script doesn't work if it's state.

Reply:
Sorry, I missed, please ignore.